### PR TITLE
🛡 Revendor `gardener/gardener@v1.39`, update Golang version to `1.17`, enable `ServiceAccount` token projection

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
* Revendor `gardener/gardener@v1.39`.
* Update Golang version to `1.17`.
* When the extension is running on a seed with a gardenlet of at least `v1.37` then `ServiceAccount` token projection is enabled.

ℹ️ The `branch.cfg` definition is still using Golang `1.16`, hence, the `verify` step fails. Once this PR is approved, we can adapt the `branch.cfg` (also update to `1.17`) and then re-trigger the CI pipeline for this PR.

**Which issue(s) this PR fixes**:
Part of gardener/gardener#4659
Part of gardener/gardener#4878

**Special notes for your reviewer**:
✅ ~Depends on https://github.com/gardener/gardener/pull/5162 and `v1.39` release, hence, PR is in draft state.~

**Release note**:

```noteworthy developer
The Golang version has been updated to `1.17.5`.
```
```noteworthy operator
The extension controller uses a projected `ServiceAccount` token in case it runs on a seed with a gardenlet of at least `v1.37` or higher.
```

### CODE REVIEW

**Comment by dimityrmirchev** [comment]
I removed some tests that were out of date and not currently used. However I am experiencing some strange behavior when I run `make generate`. I get an "invalid type" for the `healthCheckConfig` in the api-reference folder. I tried tinkering with it for a bit but with no particular luck. Is this expected or a known issue?
My go version: `go version go1.17.5 darwin/amd64`

**Comment by dimityrmirchev** [comment]
I updated the go version to `1.17.5` in branch.cfg and all seems good.

**Comment by dimityrmirchev** [comment]
@rfranzke Sorry, I did forget to submit the question I was going to ask. Can you please clarify it and I will approve the PR?

File: charts/gardener-extension-shoot-oidc-service/templates/deployment.yaml [hunk 1]
```diff
@@ -65,6 +65,9 @@ spec:
         - --webhook-config-service-port={{ .Values.webhookConfig.servicePort }}
         - --webhook-config-server-port={{ .Values.webhookConfig.serverPort }}
         - --webhook-config-namespace={{ .Release.Namespace }}
+        {{- if .Values.gardener.version }}
```
[the remark this answers is not in the document]
Comment (rfranzke) [reply charts/gardener-extension-shoot-oidc-service/templates/deployment.yaml:68]: Yes, this is correct. A new token is issued before the old token expires (~80% of the lifetime) and then re-populated at the exact same mount path. The component (OIDC extension) is responsible for regularly reloading this token from the disk (which is implicitly supported by the `client-go` version we use).

[the remark this answers is not in the document]
Comment (dimityrmirchev) [reply charts/gardener-extension-shoot-oidc-service/templates/deployment.yaml:68]: Thanks!

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**